src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int baseA;
+    private int baseB;
+    private int leg;
+    private double height;
+
+    public IsoscelesTrapezoid(int baseA, int baseB, int leg, String color) {
+        super(color);
+        this.baseA = baseA;
+        this.baseB = baseB;
+        this.leg = leg;
+        this.height = Math.sqrt(Math.pow(leg,2)
+                - Math.pow(((baseA > baseB) ? (baseA - baseB) : (baseB - baseA)),2));
```

Comment:
The height calculation assumes the trapezoid is isosceles and uses the Pythagorean theorem. However, the logic does not ensure that `baseA` is greater than `baseB`, which is necessary for this calculation to be valid. Consider adding a check or adjusting the logic to ensure `baseA` is always greater than `baseB` before performing this calculation.